### ISSUE TITLE
Tables: select feature

### ISSUE THREAD

**Issue by lianee** [bootstrap-vue/bootstrap-vue#1790]
Feature request.

I'm using Datatables and was thinking about using the Tables component instead, and there is only one thing I miss for that: ability for the user to select items, single, multiple (and the great "os" UI mode of Datatables, allowing to select multiple items using shift or ctrl/cmd keys)

[Datatables example](https://datatables.net/extensions/select/examples/initialisation/multi.html)

It is of course possible to implement it on top of the Table component, but I think it would be a great feature.

**Comment by Diaskhan** [comment]
Why checkboxes is not a option ?
https://bootstrap-vue.js.org/docs/components/table#row-details-support

Could U show workoable component, maybe some realized this kind component ?

**Comment by mosinve** [comment]
As table rows can handle clicks, you can implement your own row click handler with blackjack and multi-selecting

**Comment by mosinve** [comment]
our goal is not to provide swiss-knife-datatable-component.... we provide you a highly-customizable components, allowing to you to implement all your fantasies :)

**Comment by lianee** [comment]
You're right, and of course, the same reasoning could apply to filtering, sorting and paging ;)

What led me to write this feature request (I never said it was a show-stopper) is that implementing this feature on top of the component leads to rewrite the filter and sort parts outside the component, since the index returned by the row-clicked event is not stable, but depends on filter and sort states.

Of course, for a specific case when I know where and if each line has a unique key, it is quite easy to write without loosing filter and sort, but writing it on a more generic way might prove difficult outside the component, but maybe I missed something obvious...

Another option would be if the row-clicked event sent the original row index (then considered a key) instead or in addition of the current one

**Comment by tmorehouse** [comment]
Because of the way filtering happens within `b-table` it is almost impossible to maintain the original row key.

And if provider function + provider paging is used, we don't have the original row indexes (which the server would have defined)
